A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -2638,35 +2638,19 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     tryGoInApproachPhase() {
-        if (this.currentFlightPhase === FmgcFlightPhases.CLIMB) {
-            this.flightPhaseManager.changeFlightPhase(FmgcFlightPhases.APPROACH);
-            Coherent.call("GENERAL_ENG_THROTTLE_MANAGED_MODE_SET", ThrottleMode.AUTO);
-            SimVar.SetSimVarValue("L:A32NX_GOAROUND_PASSED", "bool", 0);
-            return true;
-        }
-        if (this.currentFlightPhase === FmgcFlightPhases.CRUISE) {
-            this.flightPhaseManager.changeFlightPhase(FmgcFlightPhases.APPROACH);
-            Coherent.call("GENERAL_ENG_THROTTLE_MANAGED_MODE_SET", ThrottleMode.AUTO);
-            SimVar.SetSimVarValue("L:A32NX_GOAROUND_PASSED", "bool", 0);
-            return true;
-        }
-        if (this.currentFlightPhase === FmgcFlightPhases.DESCENT) {
-            this.flightPhaseManager.changeFlightPhase(FmgcFlightPhases.APPROACH);
-            Coherent.call("GENERAL_ENG_THROTTLE_MANAGED_MODE_SET", ThrottleMode.AUTO);
-            SimVar.SetSimVarValue("L:A32NX_GOAROUND_PASSED", "bool", 0);
-            return true;
+        if (
+            this.currentFlightPhase === FmgcFlightPhases.PREFLIGHT ||
+            this.currentFlightPhase === FmgcFlightPhases.TAKEOFF ||
+            this.currentFlightPhase === FmgcFlightPhases.DONE
+        ) {
+            return false;
         }
-        if (this.currentFlightPhase === FmgcFlightPhases.GOAROUND) {
+
+        if (this.currentFlightPhase !== FmgcFlightPhases.APPROACH) {
             this.flightPhaseManager.changeFlightPhase(FmgcFlightPhases.APPROACH);
-            Coherent.call("GENERAL_ENG_THROTTLE_MANAGED_MODE_SET", ThrottleMode.AUTO);
-            SimVar.SetSimVarValue("L:A32NX_GOAROUND_PASSED", "bool", 0);
```

Comment:
`GENERAL_ENG_THROTTLE_MANAGED_MODE_SET` and `L:A32NX_GOAROUND_PASSED` are both set on the transition into approach phase. (onFlightPhaseChanged method)